blockchain/cryptoProvider/signer.go

```diff
@@ -0,0 +1,86 @@
+package cryptoProvider
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/ElrondNetwork/elrond-go-core/hashing/keccak"
+	"github.com/ElrondNetwork/elrond-go-crypto"
+	"github.com/ElrondNetwork/elrond-go-crypto/signing/ed25519/singlesig"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+)
+
+var (
+	signerLog     = logger.GetOrCreate("elrond-sdk-erdgo/signer")
+	hasher        = keccak.NewKeccak()
+	singleSigner  = &singlesig.Ed25519Signer{}
+	messagePrefix = []byte("\x17Elrond Signed Message:\n")
+)
+
+// signer contains the primitives used to correctly sign a transaction
+type signer struct{}
+
+// NewXSigner will create a new instance of signer
+func NewXSigner() *signer {
+	return &signer{}
+}
+
+// SignMessage will generate the signature providing the private key bytes and the message bytes
+// prepending the standard messagePrefix
+func (xs *signer) SignMessage(msg []byte, privateKey crypto.PrivateKey) ([]byte, error) {
+	serializedMessage, err := xs.serializeForSigning(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return singleSigner.Sign(privateKey, serializedMessage)
+}
+
+// VerifyMessage will verify the signature providing the public key bytes and the message bytes
+func (xs *signer) VerifyMessage(msg []byte, publicKey crypto.PublicKey, sig []byte) error {
```

Comment:
Does adding the following method `VerifyByteSlice` will improve the reusability of this component?

Reply:
Yes, I would say that if we expose `SignByteSlice` we should also expose `VerifyByteSlice`

Reply:
done